src/main/java/SearchEngine.java

```diff
@@ -46,18 +46,22 @@ private void printMenu() {
                 0. Exit""");
     }
 
-    private void findPeoples(Map<String, List<Integer>> linesIndexesByWord) throws ArrayIndexOutOfBoundsException {
+    private void findPeoples(Map<String, List<Integer>> linesIndexesByWord) {
         System.out.println("\nSelect a matching strategy: ALL, ANY, NONE");
         String strategy = scanner.nextLine();
         SearchStrategy searchStrategy = strategyChooser.get(strategy);
         System.out.println("\nEnter a name or email to search all suitable people.");
         String[] queryWords = LineUtils.convertToArray(scanner.nextLine());
-        List<String> foundPeoples = searchStrategy.search(peoples, linesIndexesByWord, queryWords);
-        if (foundPeoples.isEmpty() || queryWords[0].equals("")) {
-            System.out.println("No matching person found.");
+        if (queryWords[0].equals("")){
+            System.out.println("You pressed 'Enter'");
         } else {
-            System.out.println(String.format("\n%d persons found:", foundPeoples.size()));
-            printPeoples(foundPeoples);
+            List<String> foundPeoples = searchStrategy.search(peoples, linesIndexesByWord, queryWords);
+            if (foundPeoples.isEmpty()) {
+                System.out.println("No matching person found.");
+            } else {
+                System.out.println(String.format("\n%d persons found:", foundPeoples.size()));
+                printPeoples(foundPeoples);
+            }
         }
```

Comment:
The nested if-else does not look good, can you make this change in the `LineUtils`.

Reply:
fixed